packages/phoenix-evals/src/phoenix/evals/core/evaluators.py

```diff
@@ -0,0 +1,503 @@
+import asyncio
+import inspect
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from functools import wraps
+from string import Formatter
+from textwrap import dedent
+from typing import Any, Callable, Dict, List, Literal, Optional, Set, Tuple, Union
+
+from typing_extensions import Mapping
+
+from phoenix.evals.llm import LLM, AsyncLLM
+
+# --- Type Aliases ---
+EvalInput = Mapping[str, Any]
+Schema = Optional[Dict[str, Any]]
+SourceType = Literal["human", "llm", "heuristic"]
+DirectionType = Literal["maximize", "minimize"]
+
+ERROR_SCORE = "ERROR"
+
+
+# --- Score model ---
+@dataclass(frozen=True)
+class Score:
```

Comment:
are these scores compatible with experiment evaluators? how far away are the two as implemented?